internal/server/models/pagination.go

```diff
@@ -0,0 +1,26 @@
+package models
+
+import (
+	"math"
+)
+
+// Internal Pagination Data
+type Pagination struct {
+	Page  int
+	Limit int
+	Sort  string
+
+	TotalCount int
```

Comment:
Do we need these fields yet (or the functions below)? It doesn't seem they are called + tested anywhere, and I'd be weary of needing to have "calculated" state. We already have `Count` in our `ListResponse` type.

Reply:
This is a good point and calculating the total count/pages is less than ideal (querying a whole table twice) and can contribute to some wonky design. We do need some sort of method of communicating to the user how many requests are necessary to get the full list